### DISCUSSION TITLE
How to decompile binary that is build with Rust and compressed with upx ?

### DISCUSSION THREAD

**Discussion by ds2k5** [NationalSecurityAgency/ghidra#6922] [Q&A] [answered]
Hello,

curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
mkdir rust
cd rust
cargo new hello
cd hello
cargo check
cargo build --release
cp target/release/hello .
upx compress --brute hello
(upx 4.24 or newer needed)

cd ~ghidra_11.1.2_PUBLIC/
./ghidraRun

![grafik](https://github.com/user-attachments/assets/8d44b082-d17b-4bd2-a80c-21c7052b8433)

![grafik](https://github.com/user-attachments/assets/74592e30-9a0e-4d08-bde5-0b3cd3374f13)

![grafik](https://github.com/user-attachments/assets/b2af1fc4-22c9-49ef-8a2c-8b5556aa65e2)

![grafik](https://github.com/user-attachments/assets/10e92f73-9d1d-4379-882d-30a49a438ea8)

![grafik](https://github.com/user-attachments/assets/db591d19-7c04-4369-8ea3-334c99bc2a6e)

![grafik](https://github.com/user-attachments/assets/efd35256-787e-481b-bfa1-caac31e6f83a)

![grafik](https://github.com/user-attachments/assets/901fe5bb-70b5-4ad7-845d-54ac6be9f1db)

Using: Debian 12 with openJDK-17.0.12+7-2

What did I wrong ?

**Comment by ryanmkurtz** [comment, marked as answer]
In your Ghidra Front End GUI windows, do you see your `hello` program in the project tree?  If so, just double click it and it should open in a  new Code Browser windows.  If you don't see it, let me know.

**Reply by ds2k5** [reply]
thanks a lot! 

![grafik](https://github.com/user-attachments/assets/e6791627-1b92-4faf-905b-32c186910c69)
